### ISSUE TITLE
[Claude App] "Could not attach to MCP server github"

### ISSUE THREAD

**Issue by Lymdun** [modelcontextprotocol/servers#42]
Heya,

Currently trying to set up MCP server github on my Claude app without success - I always get a notification saying "Could not attach to MCP server github". Note that I can see my config on Settings > Developer.

mcp.logs
```txt
2024-11-25T21:27:47.135Z [info] Attempting to connect to MCP server github...
2024-11-25T21:27:47.152Z [info] Connected to MCP server github!
```

![image](https://github.com/user-attachments/assets/fc07c0fa-6959-4a00-b522-d883d1033cdc)

![image](https://github.com/user-attachments/assets/e1c437f4-860f-4b06-890a-54d850039e68)

Latest version of Claude App
OS: Windows 11

**Comment by jspahrsummers** [comment]
Are you able to get GitHub working with the [MCP Inspector](https://modelcontextprotocol.io/docs/tools/inspector)? That might help narrow down whether this is an issue in the desktop app or the server configuration.

**Comment by Lymdun** [comment]
> Are you able to get GitHub working with the [MCP Inspector](https://modelcontextprotocol.io/docs/tools/inspector)? That might help narrow down whether this is an issue in the desktop app or the server configuration.

Unfortunaly can't get MCP Inspector to work either 😅 
```txt
🔍 MCP Inspector is up and running at http://localhost:5173 🚀
[server] 'PORT' n'est pas reconnu en tant que commande interne
[server] ou externe, un programme ex�cutable ou un fichier de commandes.
[server] PORT= node C:\Users\Lymdun\AppData\Local\npm-cache\_npx\5a9d879542beca3a\node_modules\@modelcontextprotocol\inspector\server\build\index.js --env npx --args="@modelcontextprotocol/server-github GITHUB_PERSONAL_ACCESS_TOKEN=ghp_rXXXXXX" exited with code 1```

**Comment by Lymdun** [comment]
Nevermind, been able to launch it - will open an issue later in Inspector repo, could be easier to launch it from Windows (in my case, had to launch both client & server by myself using npm-cache)

![image](https://github.com/user-attachments/assets/710e19c3-6dd1-45bb-949b-14ff0b47e12f)

And the error when trying to connect: 
```bash
Error in /sse route: Error: spawn npx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npx',
  path: 'npx',
  spawnargs: [ '-y', '@modelcontextprotocol/server-github' ]
}
```

**Comment by sean-lynch** [comment]
That looks like one of the issues I was seeing here: https://github.com/modelcontextprotocol/servers/issues/37

Can you try replacing the command `npx` with the output of running `which npx` in your terminal?

**Comment by Lymdun** [comment]
Fixed by adding APPDATA varenv in inspector, now I can receive messages without issue.
Unfortunately Claude App stille have the same issue even with a fixed npx command

**Comment by Lymdun** [comment]
Just seen this log from npm-cache while toying with Claude App:

```
39 http fetch GET 200 https://registry.npmjs.org/@modelcontextprotocol%2fserver-github 1242ms (cache revalidated)
40 timing arborist:ctor Completed in 0ms
41 timing command:exec Completed in 1259ms
42 verbose stack Error: ENOENT: no such file or directory, lstat 'C:\Users\Lymdun\AppData\Local\AnthropicClaude\app-0.7.1\${APPDATA}'
42 verbose stack     at async lstat (node:internal/fs/promises:1014:18)
42 verbose stack     at async #loadActual (C:\Program Files\nodejs\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\load-actual.js:114:20)
42 verbose stack     at async exec (C:\Program Files\nodejs\node_modules\npm\node_modules\libnpmexec\lib\index.js:193:26)
42 verbose stack     at async module.exports (C:\Program Files\nodejs\node_modules\npm\lib\cli-entry.js:61:5)
43 verbose cwd C:\Users\Lymdun\AppData\Local\AnthropicClaude\app-0.7.1
44 verbose Windows_NT 10.0.22631
45 verbose node v21.6.2
46 verbose npm  v10.2.4
```

**Comment by jspahrsummers** [comment]
> Error output from MCP Server:
> 
> ```
> npm ERR! code ENOENT
> npm
> ERR!
> syscall lstat
> npm ERR! path C:\Users\Lymdun\${APPDATA}
> npm ERR! errno -4058 npm ERR! enoent ENOENT: no such file or directory, lstat 'C:\Users\Lymdun\${APPDATA}' npm ERR! enoent This is related to npm not being able to find a file. npm ERR! enoent npm ERR! A complete log of this run can be found in: C:\Users\Lymdun\npm-cache\_logs\2024-11-25T22_36_56_428Z-debug-0.log
> ```

Did this come after making changes to the server's code, by chance? I'm a bit confused about how the path in this log is being built, or what's logging `ENOENT` error messages here.

**Comment by Lymdun** [comment]
> > Error output from MCP Server:
> > ```
> > npm ERR! code ENOENT
> > npm
> > ERR!
> > syscall lstat
> > npm ERR! path C:\Users\Lymdun\${APPDATA}
> > npm ERR! errno -4058 npm ERR! enoent ENOENT: no such file or directory, ls
> Did this come after making changes to the server's code, by chance? I'm a bit confused about how the path in this log is being built, or what's logging `ENOENT` error messages here.

No, I was using the stock inspector the whole time along with the stock github-server from npm registry

**Comment by GidianB** [comment]
Im getting the same issue on my Windows machine with both brave-search and filesystem.
![image](https://github.com/user-attachments/assets/6e320b5a-4c47-4eac-9673-b1b20b63e3fc)
Im using nvm for windows if that might be of any importance.
